panel/chat/interface.py

```diff
@@ -45,6 +45,12 @@ class _ChatButtonData:
         The objects to display.
     buttons : List
         The buttons to display.
+    callback : Callable
+        The callback to execute when the button is clicked.
+    js_on_click : str | None
+        The JavaScript to execute when the button is clicked.
+    js_args : Dict[str, Any] | None
+        The JavaScript arguments to pass when the button is clicked.
```

Comment:
Think I'd prefer `js_on_click` being a Dict with 'code' and 'args' keys rather than two distinct options.